### PR TITLE
[SPARK-43383][SQL][FOLLOWUP] Use the same method to compute LocalRelation's data length.

### PR DESCRIPTION
**Pull request by yikf**
### What changes were proposed in this pull request?

As comment https://github.com/apache/spark/pull/41064#discussion_r1190332445, This PR amins to use the same method to compute LocalRelation's data length.

### Why are the changes needed?

code improve.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GA

### CODE REVIEW

**Comment by Yikf** [comment]
cc @rednaxelafx @dongjoon-hyun , Could you take a look if you find a time, thanks : )

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala [hunk 1]
```diff
@@ -80,7 +80,7 @@ case class LocalRelation(
 
   override def computeStats(): Statistics =
     Statistics(sizeInBytes = EstimationUtils.getSizePerRow(output) * data.length,
-      rowCount = Some(data.size))
+      rowCount = Some(data.length))
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala:83]: Actually, this was mentioned by @HyukjinKwon first. I thought it's addressed.
- https://github.com/apache/spark/pull/41064#discussion_r1186956202

**Review by dongjoon-hyun — APPROVED** [review]
cc @HyukjinKwon too

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you! Merged to master.

**Pull request closed without merging**